orchagent/orch.cpp

```diff
@@ -699,15 +699,38 @@ task_process_status Orch::handleSaiCreateStatus(sai_api_t api, sai_status_t stat
      *          in each orch.
      *       3. Take the type of sai api into consideration.
      */
-    switch (status)
+    switch (api)
     {
-        case SAI_STATUS_SUCCESS:
-            SWSS_LOG_WARN("SAI_STATUS_SUCCESS is not expected in handleSaiCreateStatus");
-            return task_success;
+        case SAI_API_FDB:
+            switch (status)
+            {
+                case SAI_STATUS_SUCCESS:
+                    SWSS_LOG_WARN("SAI_STATUS_SUCCESS is not expected in handleSaiCreateStatus");
```

Comment:
interesting, success is not expected :D

Reply:
Yes, this function aims to handle the failed scenarios, so it is not expected to get called for success. This is just to put a warning when this unexpected scenario happens.